website-custom-views/src/content/development/data-fetching.mdx

```diff
@@ -673,3 +673,38 @@ When using the [`@commercetools-frontend/sdk`](https://www.npmjs.com/package/@co
 ```
 
 </Info>
+
+## Fetching host application view related data
+
+You can use Custom Views to display and manage any kind of data you might need. There can be use cases where the data you want to use in your Custom View is dependant on the entity the underlying host application view is displaying.
+
+In this situations, you can use the [hostUrl](/api-reference/commercetools-frontend-application-shell-connectors#hosturl) property on the [Custom View context](/api-reference/commercetools-frontend-application-shell-connectors#custom-view-context) to get the URL of the host application where you can extract the ID of the displayed entity.
+
+Let's say your Custom View will be rendered in the order details view of the Orders application and you want to load some data related to the specific order which is being displayed in the host application.
+You can access the host application URL using the [useCustomViewContext](/api-reference/commercetools-frontend-application-shell-connectors#usecustomviewcontext) hook and extract the order ID from the URL.
+
+```jsx title="my-custom-view.jsx"
+import { useCustomViewContext } from '@commercetools-frontend/application-shell-connectors';
+
+const MyCustomView = () => {
+  const hostUrl = useCustomViewContext((context) =>
+    context.hostUrl
+  );
+
+  // The order details view URL looks like this:
+  // https://<merchant-center-domain>/<project-key>/orders/111111-2222-3333-444-5555555555/general
+  const url = new URL(hostUrl);
+  const pathnameParts = url.pathname.split('/');
+  const orderId = pathnameParts[3];
```

Comment:
Wondering if we should provide an easier way to access the route params 🤔

Reply:
Yes, that would be great although we have many different URLs in the Merchant Center so it would be difficult to maintain a clean strategy if we want to provide helpers like `getOrderIdFromUrl`, `getProductIdFromUrl`.

What do you have in mind?

Reply:
I don't have anything specific in mind but I would keep it anyway as generic as possible.

In any case, we can also wait for feedback from users.